server/app/api/blueprints/foo.py

```diff
@@ -0,0 +1,111 @@
+from flask import Blueprint, jsonify, request
+from sqlalchemy import exc
+from app import db
+from app.api.models.foo import Foo
+
+foo_blueprint = Blueprint('foo', __name__)
+
+
+@foo_blueprint.route('/api/foo', methods=['POST'], strict_slashes=False)
+def post():
+
+    response_object = {
+        'errors': ['Invalid request.']
+    }
+
+    # Handle invalid input
+    post_data = request.get_json()
+    if not post_data:
+        return jsonify(response_object), 400
+
+    # Get request data
+    string_field = post_data.get('string_field')
+
+    # Optional check if object already exists in database
+    if (len(Foo.query.filter_by(string_field=string_field).all()) > 0):
+        response_object = {
+            'errors': [f'String "{string_field}" already exists']
+        }
+        return jsonify(response_object), 400
+
+    # Add to DB
+    foo_row = Foo(string_field=string_field)
+    db.session.add(foo_row)
+    db.session.commit()
+
+    return jsonify(foo_row.to_json()), 201
+
+
+@foo_blueprint.route('/api/foo', methods=['GET'], strict_slashes=False)
+def get_all():
+    response_object = {
+        'records': [foos.to_json() for foos in Foo.query.all()]
+    }
+    return jsonify(response_object), 200
+
+
+@foo_blueprint.route('/api/foo/<int:foo_id>', methods=['GET'], strict_slashes=False)
+def get(foo_id):
+    record = Foo.query.filter_by(id=foo_id).first()
+    if not record:
+        return jsonify({
+            'errors': [f'No record with id={foo_id} found.']
```

Comment:
Missing error code? We could add a test for this case, too

Reply:
I wondered about this (and most of the things you mentioned in your comments), but I leaned on the side of leaving things "as-is" if they weren't mentioned in the ticket or included in the referenced commits from the code challenge.

So, I can make all these changes and any others if you think of them 👍

Reply:
I totally just forgot these small things lol

I must have fixed them in other commits throughout the code challenge 😓